containerd-shim-spin-config.md

```diff
@@ -0,0 +1,254 @@
+# SKIP 00X - Configuring Runtime Options in the `containerd-shim-spin`
+
+Summary: Configuring Spin runtime options in the `containerd-shim-spin`
+
+Owner: Kate Goldenring <kate.goldenring@fermyon.com> 
+
+Impacted Projects:
+
+- [x] spin-operator
+- [ ] `spin kube` plugin
+- [x] runtime-class-manager
+- [x] containerd-shim-spin
+- [ ] Governance
+- [ ] Creates a new project
+
+Created: 30/05/2024
+
+## Overview
+
+The `containerd-shim-spin` SpinKube executor currently lack the ability to configure Spin runtime flags for Spin applications. This proposal aims to address this by enabling three types of configurations: Spin execution settings, shim execution settings, and shim execution environment variables. These configurations will be set via Custom Resource Definitions (CRDs) in Kubernetes, specifically extending the SpinExecutor and SpinApp CRDs. This approach will allow operators to easily configure both the application and the platform execution environments.
```

Comment:
_lacks_

Reply:
Would it make sense to express the different kinds of configuration options from the perspective of the lifetime of the configuration? i.e. executor settings vs. application settings?

Reply:
I think this is a really good way of  distinguishing between the configuration types. I updated the document.